pkg/scheduler/scheduler.go

```diff
@@ -93,14 +93,16 @@ type Scheduler struct {
 
 	nodeInfoSnapshot *internalcache.Snapshot
 
+	// TODO: evaluate if we can remove it entirely and use per profile percentageOfNodesToScore only.
 	percentageOfNodesToScore int32
 
 	nextStartNodeIndex int
 }
 
 type schedulerOptions struct {
-	componentConfigVersion            string
-	kubeConfig                        *restclient.Config
+	componentConfigVersion string
+	kubeConfig             *restclient.Config
+	// DEPRCATED: replaced with a per profile configuration.
```

Comment:
Maybe add a warning log to notify end-users.

Reply:
> Maybe add a warning log to notify end-users.

Removed the comment with "DEPRECATED" and replaced it with `Overridden by profile level percentageOfNodesToScore`. We'll review and determine if it should be deprecated and removed in the future version. /cc @Huang-Wei

Reply:
em... maybe keep it sounds like not a bad idea. Like we want to set a non-default global value, but we do have a bit of custom configurations. Let's see what will happen.